<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PropertiesEditor</class>
 <widget class="QWidget" name="PropertiesEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>401</width>
    <height>458</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Properties</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton, QComboBox {
border-image: url(:/tulip/gui/ui/btn_26.png) 4;
border-width: 4;
padding: 0px 6px;
font-size: 12px;
}
QPushButton, QComboBox, QLabel {
color: black;
}
QPushButton:hover, QComboBox:hover {
border-image: url(:/tulip/gui/ui/btn_26_hover.png) 4;
}
QPushButton:disabled, QComboBox::disabled {
color:gray;
}
QPushButton:pressed{
border-image: url(:/tulip/gui/ui/btn_26_pressed.png) 4;
}
QPushButton::menu-indicator{
subcontrol-origin: margin;
subcontrol-position: center right;
right: 4px;
}
QPushButton {
outline: none;
margin: 2
}


QComboBox::down-arrow {
image: url(:/tulip/gui/ui/combobox_arrow.png);
}

QComboBox:drop-down {
subcontrol-origin: padding;
subcontrol-position: top right;
border-left-style: none; 
border-top-right-radius: 1px;
border-bottom-right-radius: 1px;
}

#PropertiesEditor {
background-color: white;
border: 0px;
}
QScrollArea {
border: 0px;
}
QFrame[ section=&quot;true&quot; ] {
border: 1px solid #D9D9D9;
padding-top: 0px;
}
QFrame[ subsection = &quot;true&quot; ] {
border-top: 1px solid #D9D9D9;
padding-top: 5px;
background-color: #F4F4F4;
}
#frame_2 {
border-right: 0px;
}
QPushButton[flat=&quot;true&quot;] {
font: bold;
border-image: none;
border-width: 4;
padding: 0px 6px;
font-size: 12px;
background-color: transparent;
}
QTableView {
font: 12px;
}
QTableView::indicator:checked {
image: url(:/tulip/gui/icons/eye.png);
}
QTableView::indicator:unchecked {
image: url(:/tulip/gui/icons/eye_disabled.png);
}
QCheckBox::indicator:checked {
image: url(:/tulip/gui/icons/eye.png);
}
#propsVisibilityCheck::indicator:indeterminate {
image: url(:/tulip/gui/icons/eye_partially_disabled.png);
}
QCheckBox::indicator:unchecked {
image: url(:/tulip/gui/icons/eye_disabled.png);
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frame">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="leftMargin">
       <number>5</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>42</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">font-weight: bold;
color: #595959;</string>
        </property>
        <property name="text">
         <string>&lt;font size=&quot;5&quot;&gt;Properties&lt;/font&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QCheckBox" name="propsVisibilityCheck">
          <property name="toolTip">
           <string>Show/Hide all 'Elements' table properties</string>
          </property>
          <property name="text">
           <string>Show all  properties</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="newButton">
          <property name="toolTip">
           <string>Add a new local property to the current graph</string>
          </property>
          <property name="text">
           <string>Add new</string>
          </property>
          <property name="icon">
           <iconset resource="../../../library/tulip-gui/resources/TulipGUIResource.qrc">
            <normaloff>:/tulip/gui/icons/64/list-add.png</normaloff>:/tulip/gui/icons/64/list-add.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="visualPropertiesCheck">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Show/Hide visual rendering properties</string>
        </property>
        <property name="text">
         <string>Show visual rendering properties</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="propMatchLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Only show the properties whose name
matches the given regular expression.</string>
          </property>
          <property name="text">
           <string>Show properties </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="propMatchButton">
          <property name="text">
           <string>matching</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="ClearableLineEdit" name="propertiesFilterEdit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>24</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>24</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Only show the properties whose name
matches the given regular expression.</string>
          </property>
          <property name="placeholderText">
           <string>a regular expression</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_2">
     <property name="section" stdset="0">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="margin">
       <number>0</number>
      </property>
      <item>
       <widget class="QTableView" name="tableView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>320</height>
         </size>
        </property>
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <property name="cornerButtonEnabled">
         <bool>false</bool>
        </property>
        <attribute name="horizontalHeaderMinimumSectionSize">
         <number>50</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ClearableLineEdit</class>
   <extends>QLineEdit</extends>
   <header location="global">tulip/ClearableLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../library/tulip-gui/resources/TulipGUIResource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>tableView</sender>
   <signal>customContextMenuRequested(QPoint)</signal>
   <receiver>PropertiesEditor</receiver>
   <slot>showCustomContextMenu(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>propsVisibilityCheck</sender>
   <signal>stateChanged(int)</signal>
   <receiver>PropertiesEditor</receiver>
   <slot>setPropsVisibility(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>-40</x>
     <y>108</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>visualPropertiesCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>PropertiesEditor</receiver>
   <slot>showVisualProperties(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>204</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>showCustomContextMenu(QPoint)</slot>
  <slot>showSystemProperties(bool)</slot>
  <slot>checkAll(bool)</slot>
 </slots>
</ui>
